### DISCUSSION TITLE
Does this work under linux?

### DISCUSSION THREAD

**Discussion by Plutonium5793** [XusinboyBekchanov/MyFbFramework#33] [Q&A] [answered]
I saw a comment that suggested it does,but I tried to download it and it seems to depend on windows dlls.   Is there a version that will work under linux??  Thanks,
John

**Comment by XusinboyBekchanov** [comment, marked as answer]
On Linux, you only need gtk2 or gtk3 (whichever you prefer). The current version github code and github releases also works with Linux.

**Comment by Plutonium5793** [comment]
I am not having much luck with it.  I am running Fedora 35 XFce4 with geany as the ide.  both gtk2 and gtk3 are installed.  I did finally get the hello.bas to compile successfully.  When I run the compiled version it says it cannot find frm.Center.  
I did have to modify the includes
#Include "/home/jduchek/01-FreeBasic-2022/MyFbFramework.1.3.0/mff/Form.bi"
#Include "/home/jduchek/01-FreeBasic-2022/MyFbFramework.1.3.0/mff/CommandButton.bi"
so that it would find them.  
I am assuming that frm.Center would be included in one of the .bi files so I am not certain how to get past that.  

Thanks for any help,
John

**Reply by XusinboyBekchanov** [reply]
Since version 1.2.2, frm.Center has been replaced with frm.CenterParent and frm.CenterScreen.

**Comment by Plutonium5793** [comment]
I really appreciate your help.  I replaced frm.Center with each of those and it now no longer compiles.  I get an error 18 element not defined for each of them.  Is there a manual or tutorial for the current version that I can access?  I hate to put you through all these questions....
Thanks,
John

**Reply by XusinboyBekchanov** [reply]
There are almost all properties and methods of controls:
https://github.com/XusinboyBekchanov/VisualFBEditor/discussions/98

It turns out I was wrong, they are CenterToParent and CenterToScreen:
![image](https://user-images.githubusercontent.com/32607344/159722663-5da0f424-d5fc-4df8-bc0e-a17d66fce8ed.png)

**Reply by XusinboyBekchanov** [reply]
Fixed: Hello Example compiling:
https://github.com/XusinboyBekchanov/MyFbFramework/commit/b263978314b4237254ce60264ac39fe1fa2ecaf8
